src/go/cmd/module.go

```diff
@@ -114,94 +114,114 @@ func NewModule(dir string) (*Module, error) {
 	// the definition from azcore/internal/shared into the APIView for azcore, making the type's
 	// fields visible there.
 	externalPackages := map[string]*Pkg{}
+
+	// tracks which packages have had their type aliases resolved.
+	// this prevents resolving dependent packages multiple times.
+	processedPackages := map[string]struct{}{}
+
 	for _, p := range m.packages {
-		for alias, qn := range p.typeAliases {
```

Comment:
This content was moved into `recursiveResolveTypeAliases`. The only changes to it are the recursive call to `recursiveResolveTypeAliases` and checking/updating `processedPackages`.